bin/spack

```diff
@@ -70,6 +71,21 @@ if "ruamel.yaml" in sys.modules:
 if "ruamel" in sys.modules:
     del sys.modules["ruamel"]
 
+# The following code is here to avoid failures when updating
+# the develop version, due to spurious argparse.pyc files remaining
+# in the libs/spack/external directory, see:
+# https://github.com/spack/spack/pull/25376
+# TODO: Remove in v0.18.0 or later
+try:
+    import argparse
+except ImportError:
+    argparse_pyc = os.path.join(spack_external_libs, 'argparse.pyc')
+    if not os.path.exists(argparse_pyc):
+        raise
```

Comment:
I think this should also check whether `os.access(argparse_pyc, os.W_OK)`, and if it's not writable (deletable) then we should print some message about asking your sysadmin to run `spack clean -p`.  Right now it'll end up raising if the user doesn't have access to remove the file.

Reply:
Also there is an example of how to easily test this here: https://stackoverflow.com/a/28361013/9122

I think you can:
1. Move this to `spack.main`
2. Stick it all in a function like `check_argparse_import()` and have it return the module
3. replace the import in main with `argparse = check_argparse_import()`

Then you can just call `check_argparse_import()` in a test to test it.  It'll be easier to remove later this way too.

Reply:
I suppose you could also just make it a function and call it from here in the spack script where it is, and leave the `argparse` import in `main` alone.

Reply:
I'm fine with both solutions (make it a function here or in `spack.main`). I put it here since it's where we modify `sys.path` and the code seems related.

Reply:
> Also there is an example of how to easily test this here: https://stackoverflow.com/a/28361013/9122

I think it's more complicated than this, as that example imports two different names (`foo_fast` and `foo`). Here we are importing the same name in the `try` and the `except` block, and we try to delete leftover bytecode between the two tries. I can mock something up but it will probably be more code than the entire PR. Let me know if you want me to proceed with that.

Reply:
LGTM -- thanks!